rig/machine_control/bmp_controller.py

```diff
@@ -168,9 +168,8 @@ def _send_scp(self, cabinet, frame, board, *args, **kwargs):
 
         return connection.send_scp(length, 0, 0, board, *args, **kwargs)
 
-    @ContextMixin.use_contextual_arguments
-    def get_software_version(self, cabinet=Required, frame=Required,
-                             board=Required):
+    @ContextMixin.use_contextual_arguments()
```

Comment:
With the minor issue that suddenly now you can confuse yourself (silently) with:

	@use_contextual_arguments(my_mandatory_kw_only_argument)
	def f(**kwargs):
		pass

But anyway, when a problem with decorators arrises, just add decorators!

def maybe_args_decorator(d):
	@functools.wraps(d)
	def f_(*args, **kwargs):
		if len(args) == 1 and hasattr(args[0], "__call__") and not kwargs:
			return d()(args[0])
		else:
			return d(*args, **kwargs)

(Untested)